src/Chessboard.cpp

```diff
@@ -0,0 +1,313 @@
+#include "Chessboard.h"
+#include "Rook.h"
+#include "Bishop.h"
+#include "King.h"
+#include "Queen.h"
+#include "Knight.h"
+#include "Pawn.h"
+
+Chessboard::Chessboard() {
+    initializeBoard();
+    currentPlayer_ = PieceColor::White;
+}
+
+
+Chessboard::~Chessboard() {
+    clearBoard();
+}
+
+void Chessboard::initializeBoard() {
+    for (int row = 0; row < 8; row++) {
+        for (int col = 0; col < 8; col++) {
+            board_[row][col] = nullptr;
+        }
+    }
+
+    // initialize white pieces
+    board_[0][0] = new Rook(PieceColor::White, 0, 0);
+    board_[0][1] = new Knight(PieceColor::White, 0, 1);
+    board_[0][2] = new Bishop(PieceColor::White, 0, 2);
+    board_[0][3] = new Queen(PieceColor::White, 0, 3);
+    board_[0][4] = new King(PieceColor::White, 0, 4);
+    board_[0][5] = new Bishop(PieceColor::White, 0, 5);
+    board_[0][6] = new Knight(PieceColor::White, 0, 6);
+    board_[0][7] = new Rook(PieceColor::White, 0, 7);
+
+    for (int i = 0; i < 8; i++) {
+        board_[1][i] = new Pawn(PieceColor::White, 1, i);
+    }
+
+    // initialize black pieces
+    board_[7][0] = new Rook(PieceColor::Black, 7, 0);
+    board_[7][1] = new Knight(PieceColor::Black, 7, 1);
+    board_[7][2] = new Bishop(PieceColor::Black, 7, 2);
+    board_[7][3] = new Queen(PieceColor::Black, 7, 3);
+    board_[7][4] = new King(PieceColor::Black, 7, 4);
+    board_[7][5] = new Bishop(PieceColor::Black, 7, 5);
+    board_[7][6] = new Knight(PieceColor::Black, 7, 6);
+    board_[7][7] = new Rook(PieceColor::Black, 7, 7);
+
+    for (int i = 0; i < 8; i++) {
+        board_[6][i] = new Pawn(PieceColor::Black, 6, i);
+    }
+}
+
+
+void Chessboard::clearBoard() {
+    for (int i = 0; i < 8; i++) {
+        for (int j = 0; j < 8; j++) {
+            if (board_[i][j] != nullptr) {
+                delete board_[i][j];
+            }

```

Comment:
If the board_ member variable should not be directly accessible outside the Chessboard class, you can encapsulate it by providing public member functions to manipulate the board, such as a removePiece function.